Sources/Onboarding/MainCoordinator.swift

```diff
@@ -4,26 +4,26 @@ import Logging
 import UIKit
 
 final class MainCoordinator: NSObject,
+                             ParentCoordinator,
                              NavigationCoordinator {
-    private let window: UIWindow
     let root: UINavigationController
     let session: LoginSession
     let analyticsService: AnalyticsService
+    var childCoordinators = [ChildCoordinator]()
     private let viewControllerFactory = OnboardingViewControllerFactory.self
     
-    init(window: UIWindow,
-         root: UINavigationController,
+    init(root: UINavigationController,
          session: LoginSession,
          analyticsService: AnalyticsService = OneLoginAnalyticsService()) {
-        self.window = window
         self.root = root
         self.session = session
         self.analyticsService = analyticsService
     }
     
     func start() {
-        let introViewController = viewControllerFactory.createIntroViewController(analyticsService: analyticsService,
-                                                                                  session: session)
+        let introViewController = viewControllerFactory.createIntroViewController(analyticsService: analyticsService) { [self] in
```

Comment:
It's not something for this pull request, but I've chatted with Robin on some thoughts around flow when opening the app: worth having a catch up with him around the way he's done similar things for GOV.UK.

I think our end-goal is:
- MainCoordinator which is a `TabBarCoordinator`.
- Tabs with child coordinators for Home, Wallet, etc.
- `MainCoordinator` opens the `AuthenticationCoordinator` flow modally (full-screen) on it's root (UITabBarController)

Reply:
Interesting, will do!
My thoughts were very much down the line of a MainCoordinator which coordinates the other coordinators... 😂 
So essentially the MainCoordinator handles the Welcome screen - starts an AuthenticationCoordinator for Sign Up or Log In then finishes when that's done - then starts a TabBarCoordinator or something suchlike for the main function of the app.
Sounds a bit different to what you've mentioned above so yeah might be good for a wider conversation about that 🙂